pgmpy/models/DiscreteBayesianNetwork.py

```diff
@@ -1181,6 +1181,16 @@ def do(self, nodes, inplace=False):
         else:
             nodes = list(nodes)
 
+        if isinstance(states, (str, int)):
```

Comment:
I think these checks will get quite a bit simplified after accepting dict instead of two arguments.